Shoko.CLI/Program.cs

```diff
@@ -1,89 +1,74 @@
+        => configurationBuilder.AddEnvironmentVariables(prefix: "DOTNET_")
+                               .AddCommandLineIfNotNull();
 
-            builder.UseContentRoot(Directory.GetCurrentDirectory());
-            builder.ConfigureHostConfiguration(config =>
-            {
-                config.AddEnvironmentVariables(prefix: "DOTNET_");
-                if (args != null)
-                {
-                    config.AddCommandLine(args);
-                }
-            });
-
-            builder.ConfigureAppConfiguration((hostingContext, config) =>
-            {
-                var env = hostingContext.HostingEnvironment;
-                
-                var reloadOnChange = hostingContext.Configuration.GetValue("hostBuilder:reloadConfigOnChange", defaultValue: true);
+    private static void ConfigureApp(HostBuilderContext hostingContext, IConfigurationBuilder configurationBuilder)
+    {
+        var shouldReloadOnChange = hostingContext.Configuration.GetValue("hostBuilder:reloadConfigOnChange", defaultValue: true);
+        configurationBuilder.AddJsonFile("appsettings.json",
+                                         optional: true,
+                                         reloadOnChange: shouldReloadOnChange)
+                            .AddJsonFile($"appsettings.{hostingContext.HostingEnvironment.EnvironmentName}.json",
+                                         optional: true,
+                                         reloadOnChange: shouldReloadOnChange)
+                            .AddDevelopmentAssembly(hostingContext.HostingEnvironment)
+                            .AddEnvironmentVariables()
+                            .AddCommandLineIfNotNull();
+    }
 
-                config.AddJsonFile("appsettings.json", optional: true, reloadOnChange: reloadOnChange)
-                      .AddJsonFile($"appsettings.{env.EnvironmentName}.json", optional: true, reloadOnChange: reloadOnChange);
+    private static void ConfigureDefaultServiceProvider(HostBuilderContext context, ServiceProviderOptions options)
+    {
+        var isDevelopment = context.HostingEnvironment.IsDevelopment();
+        options.ValidateScopes  = isDevelopment;
+        options.ValidateOnBuild = isDevelopment;
+    }
 
-                if (env.IsDevelopment() && !string.IsNullOrEmpty(env.ApplicationName))
-                {
-                    var appAssembly = Assembly.Load(new AssemblyName(env.ApplicationName));
-                    if (appAssembly != null)
-                    {
-                        config.AddUserSecrets(appAssembly, optional: true);
-                    }
-                }
+    private static void ConfigureServices(IServiceCollection services) => services.AddHostedService<Worker>();
 
-                config.AddEnvironmentVariables();
+#region IConfigurationBuilderExtensions
+    private static IConfigurationBuilder AddCommandLineIfNotNull(this IConfigurationBuilder configurationBuilder)
```

Comment:
Correct me if I'm wrong. This allows the existing command line stuff to work, but doesn't bind to the intermediate model like it used to. Is that right?

Reply:
I'm not quite sure which model youre talking about. If you mean the ``services.AddSingleton(arguments);`` Singleton which did only contain the instance name, then yes, that is gone.

Reply:
Mk that's fine